### ISSUE TITLE
RecursionError with get_all_items when catalog is a file

### ISSUE THREAD

**Issue by sdtaylor** [stac-utils/pystac-client#607]
I was trying to list items on some catalogs and ran into this error. I thought it was an issue with numerous catalogs within catalogs, but it happens with the example catalog as well.

```
import pystac_client
print(f'pystac_client version : {pystac_client.__version__}')

import pystac
print(f'pystac version : {pystac.__version__}')

example_catalog = pystac_client.Client.from_file('https://raw.githubusercontent.com/stac-utils/pystac/main/docs/example-catalog/catalog.json')
for i in example_catalog.get_all_items():
    pass
```

```
pystac_client version : 0.7.5
pystac version : 1.8.4
/home/shawn/miniconda3/envs/geo/lib/python3.8/site-packages/pystac_client/client.py:456: FallbackToPystac: Falling back to pystac. This might be slow.
  self._warn_about_fallback("ITEM_SEARCH")
---------------------------------------------------------------------------
RecursionError                            Traceback (most recent call last)
<ipython-input-1-c935054a4da2> in <module>
      6 
      7 example_catalog = pystac_client.Client.from_file('https://raw.githubusercontent.com/stac-utils/pystac/main/docs/example-catalog/catalog.json')
----> 8 for i in example_catalog.get_all_items():
      9     pass
     10 

~/miniconda3/envs/geo/lib/python3.8/site-packages/pystac_client/client.py in get_all_items(self)
    468                 child links.
    469         """
--> 470         yield from self.get_items()
    471 
    472     def search(

~/miniconda3/envs/geo/lib/python3.8/site-packages/pystac_client/client.py in get_items(self, recursive, *ids)
    455         else:
    456             self._warn_about_fallback("ITEM_SEARCH")
--> 457             for item in super().get_items(*ids, recursive=True):
    458                 call_modifier(self.modifier, item)
    459                 yield item

~/miniconda3/envs/geo/lib/python3.8/site-packages/pystac/catalog.py in get_items(self, recursive, *ids)
    533             yield from (i for i in items if i.id in ids)
    534         else:
--> 535             yield from items
    536 
    537     def clear_items(self) -> None:

... last 2 frames repeated, from the frame below ...

~/miniconda3/envs/geo/lib/python3.8/site-packages/pystac_client/client.py in get_items(self, recursive, *ids)
    455         else:
    456             self._warn_about_fallback("ITEM_SEARCH")
--> 457             for item in super().get_items(*ids, recursive=True):
    458                 call_modifier(self.modifier, item)
    459                 yield item

RecursionError: maximum recursion depth exceeded in comparison

```

**Comment by TomAugspurger** [comment]
This probably shouldn't throw a RecursionError, but I am curious about your use-case here. `pystac-client` is focused on working with STAC APIs. Are you able to use just `pystac.read_file` instead?

**Comment by sdtaylor** [comment]
@TomAugspurger , yep, using `pystac.read_file` and `get_all_items` works in the above example.

**Comment by gadomski** [comment]
Fixed by #608.
